recipes/xmlsec/all/conanfile.py

```diff
@@ -0,0 +1,96 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+import os
+
+class XmlSecConan(ConanFile):
+    name = "xmlsec"
+    description = "XML Security Library is a C library based on LibXML2. The library supports major XML security standards."
+    license = ("MIT", "MPL-1.1")
+    homepage = "https://github.com/lsh123/xmlsec"
+    url = "https://github.com/conan-io/conan-center-index"
+    generators = "pkg_config"
+    settings = "os", "compiler", "arch", "build_type"
+    topics = ("xml", "signature", "encryption")
+    options = {
+        "shared": [True, False], 
+        "fPIC": [True, False],
+        "with_xslt": [True, False]
+        }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_xslt": False
+        }
+    
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        self.requires("libxml2/2.9.10")
+        self.requires("openssl/1.1.1h")
+        if self.options.with_xslt:
+            self.requires("libxslt/1.1.34")
+    
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/20200517")
+        self.build_requires("libtool/2.4.6")
+        self.build_requires("pkgconf/1.7.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "xmlsec-xmlsec-{}".format(self.version.replace('.', '_'))
+        os.rename(extracted_folder, self._source_subfolder)
+    
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        yes_no = lambda v: "yes" if v else "no"
+        configure_args = [
+            "--enable-crypto-dl={}".format(yes_no(False)),
+            "--enable-apps-crypto-dl={}".format(yes_no(False)),
+            "--with-libxslt={}".format(yes_no(self.options.with_xslt)),
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+        ]
+        self._autotools.configure(args=configure_args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            self.run("autoreconf -fiv", run_environment=True, win_bash=tools.os_info.is_windows)
```

Comment:
@steinerthomas 
automake uses `AUTOMAKE_CONAN_INCLUDES` and `ACLOCAL` environment variables to pass extra paths to `m4` scripts.
On posix systems, multiple paths are joined using `:`.
On Windows, they are joined by `;`.

This is essentially a missing conan feature.
I don't think I've opened an issue for this yet.

```suggestion
        # FIXME: conan is missing a feature to change the join character on lists of environment variables
        # e.g. self.env_info.AUTOMAKE_CONAN_INCLUDES.joiner = ":"
        with tools.environment_append({"AUTOMAKE_CONAN_INCLUDES": tools.get_env("AUTOMAKE_CONAN_INCLUDES", "").replace(";", ":")}):
            with tools.chdir(self._source_subfolder):
                self.run("autoreconf -fiv", run_environment=True, win_bash=tools.os_info.is_windows)
```